Object of type 'IPv4Network' is not JSON serializable

## Report

- [x] Is conjure-up running inside a virtual machine?
yes
- [x] Is this problem reproducible with `sudo snap refresh conjure-up --edge`?
yes

## Please provide the output of the following commands

```
which juju
2.3-alpha1-xenial-amd64

which conjure-up
conjure-up (edge) 2.3-alpha1-20170830.2008

which lxc
config:
  core.https_address: '[::]:8443'
  core.trust_password: true

lxc version
2.17
```

Please attach tarball of **~/.cache/conjure-up**:

```
2017-08-31 17:53:23,346 [DEBUG] conjure-up/canonical-kubernetes - utils.py:599 - Parsing 10: lxdbr0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UNKNOWN group default qlen 1000
    link/ether f6:77:b0:7d:76:13 brd ff:ff:ff:ff:ff:ff
    inet 10.206.188.1/24 scope global lxdbr0
       valid_lft forever preferred_lft forever
    inet6 fe80::f477:b0ff:fe7d:7613/64 scope link
       valid_lft forever preferred_lft forever
 for IPv4 address
2017-08-31 17:53:23,350 [DEBUG] conjure-up/canonical-kubernetes - events.py:53 - Setting Error at conjureup/events.py:147
2017-08-31 17:53:23,357 [ERROR] conjure-up/canonical-kubernetes - events.py:159 - Unhandled exception in <Task finished coro=<LXDSetupController.set_lxd_info() done, defined at /snap/conjure-up/674/lib/python3.6/site-packages/conjureup/controllers/lxdsetup/gui.py:24> exception=TypeError("Object of type 'IPv4Network' is not JSON serializable",)>
Traceback (most recent call last):
  File "/snap/conjure-up/674/lib/python3.6/site-packages/conjureup/controllers/lxdsetup/gui.py", line 29, in set_lxd_info
    self.set_state('lxd-network', iface.network)
  File "/snap/conjure-up/674/lib/python3.6/site-packages/conjureup/controllers/lxdsetup/common.py", line 31, in set_state
    ret = app.state.set(key, value)
  File "/snap/conjure-up/674/lib/python3.6/site-packages/charmhelpers/core/unitdata.py", line 274, in set
    serialized = json.dumps(value)
  File "/snap/conjure-up/674/usr/lib/python3.6/json/__init__.py", line 231, in dumps
    return _default_encoder.encode(obj)
  File "/snap/conjure-up/674/usr/lib/python3.6/json/encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/snap/conjure-up/674/usr/lib/python3.6/json/encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
  File "/snap/conjure-up/674/usr/lib/python3.6/json/encoder.py", line 180, in default
    o.__class__.__name__)
TypeError: Object of type 'IPv4Network' is not JSON serializable
2017-08-31 17:53:23,388 [DEBUG] conjure-up/canonical-kubernetes - __init__.py:21 - Showing dialog for exception: Object of type 'IPv4Network' is not JSON serializable
2017-08-31 17:55:25,570 [DEBUG] conjure-up/canonical-kubernetes - events.py:53 - Setting Shutdown at conjureup/events.py:143

```

## What Spell was Selected?
Canonical Kubernetes 

## What provider (aws, maas, localhost, etc)?
localhost 


## Additional Information

--- a/conjureup/controllers/lxdsetup/common.py
+++ b/conjureup/controllers/lxdsetup/common.py
@@ -28,12 +28,12 @@ class BaseLXDSetupController:
 
     def set_state(self, key, value):
         key = "{}.{}".format(self.state_key, key)
-        ret = app.state.set(key, value)
+        ret = app.state.set(key, str(value))
         return ret
 
     def get_state(self, key):
         key = "{}.{}".format(self.state_key, key)
-        return app.state.get(key).decode('utf8')
+        return app.state.get(key)
 
     def next_screen(self):
         return controllers.use('controllerpicker').render()

